### PR TITLE
filter playspace feature added

### PR DESCRIPTION
**Pull request by Grub2801**

### CODE REVIEW

File: app/views/layouts/_sidebar.html.erb [hunk 1]
```diff
@@ -20,7 +20,63 @@
...
+            <%= f.check_box :snack_shop, {}, true, false  %>
+          </div>
+          <div>
+            <label>toilet</label>
+            <%= f.check_box :wc, {}, true, false  %>
+          </div>
+        </div>
+        <%= submit_tag 'Search', class:"btn btn-default" %>
+      <% end %>
+    </div>
```
Comment (dshcheung) [inline app/views/layouts/_sidebar.html.erb]: can you put this filter form into a partial

File: app/assets/javascripts/playspaces.js [hunk 2]
```diff
@@ -7,27 +7,35 @@ var PlayspaceSidebar = function () {
...
-    if ($("#playspace_region").val() == "HK") {
-      for (var i = 0; i < option["HK"].length; i++) {
-        $("#playspace_district").append("<option value='"+option["HK"][i]+"'>"+option["HK"][i]+"</option>")
+  var bindRegionDistrict = function (formID) {
+    $(formID + " #playspace_region").on('change', function() {
+      $(formID + " #playspace_district").html("");
+      $(formID + " #playspace_district").append("<option value>Select by district:</option>")
+
+      if ($(formID + " #playspace_region").val() == "HK") {
+        for (var i = 0; i < option["HK"].length; i++) {
```
Comment (f3r) [inline app/assets/javascripts/playspaces.js]: `foreach` pls!

File: app/controllers/playspaces_controller.rb [hunk 3]
```diff
@@ -38,4 +37,49 @@ def playspaces_params
...
+  end
+
+  def filter_queries(results=nil)
+    require 'wannabe_bool'
+    results = results || Playspace.all
+
+    if params[:playspace] != nil
+      puts "started Filtering"
+      f_params = filter_params
+      name = f_params[:name] == nil || f_params[:name] == "" ? nil : f_params[:name].downcase
```
Comment (f3r) [inline app/controllers/playspaces_controller.rb]: indentation, be nice piiip!

File: app/assets/javascripts/playspaces.js (hunk 2 above)
Comment (Grub2801) [inline app/assets/javascripts/playspaces.js]: like this?
formID.each do |f|
?

File: app/controllers/playspaces_controller.rb (hunk 3 above)
Comment (Grub2801) [inline app/controllers/playspaces_controller.rb]: $(document).ready(function(){
  var option = {
    HK: ["Central", "North Point", "Repulse Bay", "The Peak", "Wan Chai", "Western"],
    Kln: ["Ho Man Tin", "Tsim Sha Tsui", "Wong Tai Sin"],
    NT: ["Sai Kung", "Sha Tin", "Tai Po"]
  }
  
var bindRegionDistrict = function (formID) {

  $(formID + " #playspace_region").on('change', function() {
    $(formID + " #playspace_district").html("");
    $(formID + " #playspace_district").append("<option value>Select by district:</option>")


    option.each |r|
      for (var i = 0; i < option[r].length; i++) {
        $(formID + " #playspace_district").append("<option value='"+option[r][i]+"'>"+option[r][i]+"</option>")
      }
    }

  })
};

  bindRegionDistrict('#search')
  bindRegionDistrict('#new_playspace')

})

File: app/assets/javascripts/playspaces.js (hunk 2 above)
Comment (Grub2801) [inline app/assets/javascripts/playspaces.js]: var bindRegionDistrict = function (formID) {

  $(formID + " #playspace_region").on('change', function() {
    $(formID + " #playspace_district").html("");
    $(formID + " #playspace_district").append("<option value>Select by district:</option>")


    option.each |r|
      for (var i = 0; i < option[r].length; i++) {
        $(formID + " #playspace_district").append("<option value='"+option[r][i]+"'>"+option[r][i]+"</option>")
      }
    }

  })
};

  bindRegionDistrict('#search')
  bindRegionDistrict('#new_playspace')

})

**Pull request merged**